In [1]:
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy.interpolate import interp1d
from featureExtraction import extract_grasp_polygons_over_time

In [ ]:
df = pd.read_csv("processed/user0/User0_TransformLog_SmallCube_Grasp_config3.csv")
features= extract_grasp_polygons_over_time(df)
print(features.shape)
print(features.head())


(197, 31)
   Timestamp  tip_edge0_x  tip_edge0_y  tip_edge0_z  tip_edge1_x  tip_edge1_y  \
0      37.63         4.96        -0.57         6.19         2.60        -0.13   
1      37.64         4.96        -0.57         6.18         2.59        -0.13   
2      37.65         4.96        -0.55         6.19         2.59        -0.13   
3      37.67         4.96        -0.55         6.19         2.59        -0.13   
4      37.68         4.96        -0.54         6.19         2.60        -0.13   

   tip_edge1_z  tip_edge2_x  tip_edge2_y  tip_edge2_z  ...  \
0         0.75         1.98         0.04        -1.40  ...   
1         0.76         1.99         0.05        -1.41  ...   
2         0.75         2.00         0.04        -1.40  ...   
3         0.75         1.99         0.05        -1.41  ...   
4         0.75         2.00         0.04        -1.41  ...   

   intermediate_edge1_z  intermediate_edge2_x  intermediate_edge2_y  \
0                  0.17                  1.93              

features shape meaning:

| `197`     | Rows    | We have **197 timestamps (frames)** — i.e., 197 moments during this one grasp trial where features were extracted.

| `31`      | Columns | We have **31 features per frame**: 30 edge vector features + 1 timestamp.      
